sdk/spring/azure-spring-boot/src/test/java/com/azure/spring/aad/webapp/AADAccessTokenGroupRolesExtractionTest.java

```diff
@@ -126,35 +131,58 @@ public void testEnableFullList() {
 
         Mockito.lenient().when(graphClient.getGroupsFromGraph(accessToken.getTokenValue()))
-               .thenReturn(allowedGroupIdsAndGroupNames);
-        Set<String> groupIds = userService.extractGroupRolesFromAccessToken(accessToken);
-        assertThat(groupIds).hasSize(4);
+               .thenReturn(allowedGroup);
+        Set<String> groups = userService.extractGroupRolesFromAccessToken(accessToken);
+        assertThat(groups).hasSize(3);
+        assertThat(groups).contains("ROLE_group1");
     }
 
     @Test
     public void testDisableFullList() {
-        Set<String> allowedGroupIdsAndGroupNames = new HashSet<>();
-        allowedGroupIdsAndGroupNames.add("d07c0bd6-4aab-45ac-b87c-23e8d00194ab");
-        allowedGroupIdsAndGroupNames.add("6eddcc22-a24a-4459-b036-b9d9fc0f0bc7");
-        allowedGroupIdsAndGroupNames.add("group1");
-        allowedGroupIdsAndGroupNames.add("group2");
+        allowedGroupIds.add("d07c0bd6-4aab-45ac-b87c-23e8d00194ab");
+        allowedGroupIds.add("6eddcc22-a24a-4459-b036-b9d9fc0f0bc7");
+        allowedGroupNames.add("group1");
+        allowedGroupNames.add("group2");
+
+        List<String> customizeGroupNames = new ArrayList<>();
+        Set<String> customizeGroupIds = new HashSet<>();
+        customizeGroupIds.add("d07c0bd6-4aab-45ac-b87c-23e8d00194ab");
+        customizeGroupNames.add("group1");
 
         userGroup.setEnableFullList(false);
-        Set<String> allowedGroupIds = new HashSet<>();
+        userGroup.setAllowedGroupIds(customizeGroupIds);
+        userGroup.setAllowedGroupNames(customizeGroupNames);
+
+        Mockito.lenient().when(graphClient.getGroupsFromGraph(accessToken.getTokenValue()))
+               .thenReturn(allowedGroup);
+        Set<String> groups = userService.extractGroupRolesFromAccessToken(accessToken);
+
+        assertThat(groups).contains("ROLE_group1");
+        assertThat(groups).doesNotContain("ROLE_group5");
+        assertThat(groups).contains("ROLE_d07c0bd6-4aab-45ac-b87c-23e8d00194ab");
+        assertThat(groups).doesNotContain("ROLE_d07c0bd6-4aab-45ac-b87c-23e8d00194abaaa");
+        assertThat(groups).hasSize(2);
+    }
+
+    @Test
+    public void testUseWildcards() {
         allowedGroupIds.add("d07c0bd6-4aab-45ac-b87c-23e8d00194ab");
-        userGroup.setAllowedGroupIds(allowedGroupIds);
-        List<String> allowedGroupNames = new ArrayList<>();
+        allowedGroupIds.add("6eddcc22-a24a-4459-b036-b9d9fc0f0bc7");
         allowedGroupNames.add("group1");
-        userGroup.setAllowedGroupNames(allowedGroupNames);
+        allowedGroupNames.add("group2");
+
+        Set<String> customizeGroupIds = new HashSet<>();
+        customizeGroupIds.add("all");
+        List<String> customizeGroupName = new ArrayList<>();
+        customizeGroupName.add("group1");
+
+        userGroup.setAllowedGroupIds(customizeGroupIds);
+        userGroup.setAllowedGroupNames(customizeGroupName);
+        userGroup.setEnableFullList(true);
```

Comment:
It's better to split into 2 tests: with and without `userGroup.setEnableFullList(true);`.